### ISSUE TITLE
Should not report CS8602 for extension method delegate with nullable receiver

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#30563]
Should report CS8602 warning for the instance method but not the extension method:
```c#
[module: System.Runtime.CompilerServices.NonNullTypes]

delegate void D();

class C
{
    void F1() { }
    static void F(C? x, C? y)
    {
        D d;
        d = x.F1; // warning
        d = y.F2; // ok
    }
}

static class E
{
    internal static void F2(this C? c) { }
}
```

**Comment by jcouv** [comment]
If we decide to do this, we should probably also open an issue for invocations of extension methods (`static void M(this C? c)`).
